File: examples/ruby/guides/ai_agents/mock_agent.rb
# frozen_string_literal: true

# Mock LLM and tools - no external API dependencies
@llm_call_count = 0

def call_llm(messages)
  @llm_call_count += 1
  if @llm_call_count == 1
    { "content" => "", "tool_calls" => [{ "name" => "get_weather", "args" => { "location" => "SF" } }], "done" => false }
  else
    { "content" => "It's 72°F and sunny in SF.", "tool_calls" => [], "done" => true }
  end
end

def run_tool(name, args)
  if name == "get_weather"
    loc = args["location"] || "unknown"
    "Weather in #{loc}: 72°F, sunny"
  else
    "Unknown tool: #{name}"
  end
end

File: examples/python/guides/ai_agents/mock_agent.py
"""Mock LLM and tools - no external API dependencies."""

_call_count: dict[str, int] = {}


def call_llm(messages: list[dict]) -> dict:
    """Mock LLM: first call returns tool_calls, second returns final answer."""
    key = "default"
    _call_count[key] = _call_count.get(key, 0) + 1
    if _call_count[key] == 1:
        return {
            "content": "",
            "tool_calls": [{"name": "get_weather", "args": {"location": "SF"}}],
            "done": False,
        }
    return {"content": "It's 72°F and sunny in SF.", "tool_calls": [], "done": True}


def run_tool(name: str, args: dict) -> str:
    """Mock tool execution - returns canned results."""
    if name == "get_weather":
        loc = args.get("location", "unknown")
        return f"Weather in {loc}: 72°F, sunny"
    return f"Unknown tool: {name}"
